include/SFML/Graphics/Text.hpp

```diff
@@ -378,6 +378,26 @@ class SFML_GRAPHICS_API Text : public Drawable, public Transformable
     ////////////////////////////////////////////////////////////
     Vector2f findCharacterPos(std::size_t index) const;
 
+    ////////////////////////////////////////////////////////////
+    /// \brief Calculates if text will fit within a given AABB
+    ///
+    /// Iterates through every character in the string and tests
+    /// that they will fit within \b bounds. Each character is
+    /// fully transformed for this test, so translation,
+    /// rotation, scale and origin are considered.
+    /// When any character exceeds the bounds, \a oob will be
+    /// valid. If every character is contained, \a oob will be
+    /// set to the drawn character count.
+    ///
+    /// \param bounds Bounding box that all drawn characters will be tested against
+    ///
+    /// \param oob Index of the first out-of-bounds character
+    ///
+    /// \return true if every drawn character will appear within \a bounds
+    ///
+    ////////////////////////////////////////////////////////////
+    bool fitsWithinBounds(const sf::FloatRect& bounds, std::size_t& oob) const;
```

Comment:
Is there any way to change this interface such that it doesn't require using an out parameter?

Reply:
What is the preferred SFML way to deal with optional returns? A trivial struct, std::optional<>, etc? Or perhaps just returning oob always would be a better option?

I'm not super familiar with SFML's preference for this, so any advice to meet code style is appreciated.

Reply:
`std::optional<std::size_t>` return type would probably be the best option.

Name should then be changed accordingly: `firstOutOfBoundsChar` or `firstOobCharIndex` or so.